lib/collection/src/collection/query.rs

```diff
@@ -217,7 +217,12 @@ impl Collection {
         .await?;
 
         // update timeout
-        let timeout = timeout.map(|timeout| timeout.saturating_sub(start.elapsed()));
+        let timeout = timeout.map(|timeout| {
+            // timeout is calculated with sufficient precision and avoids flooring
+            (timeout.as_secs_f32() - start.elapsed().as_secs_f32())
+                .ceil()
+                .max(1.0) as usize
```

Comment:
I think this would be better:

```suggestion
            // avoid flooring the timeout to 0
            (timeout.saturating_sub(start).as_secs_f32().ceil() as usize
```

We prevent going below zero. It'll only be 0 if enough time has already passed, and otherwise it's always at least 1.